src/docfx/lib/log/ErrorList.cs

```diff
@@ -10,15 +10,23 @@ namespace Microsoft.Docs.Build
 {
     internal class ErrorList : ErrorBuilder, IReadOnlyList<Error>
     {
+        private readonly object _lock = new();
+
         private readonly List<Error> _items = new();
 
-        public Error this[int index] => _items[index];
+        public Error this[int index] => _items.ToArray()[index];
```

Comment:
Must guard every access to `_items` with lock. You also don't need a dedicated `_lock` object, can simply lock `_items`.

Reply:
```csharp
class ErrorList : ErrorBuilder
{
    public void Error[] ToArray();
}
```